src/main/java/mate/academy/dao/BookDaoImpl.java

```
@@ -0,0 +1,116 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find book by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> bookList = mapToBook(resultSet);
+            return bookList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update the book " + book.getTitle(), e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not delete book by id " + id, e);
+        }
+    }
+
+    private List<Book> mapToBook(ResultSet resultSet) {
```

you can throw an exception in signature